### Importing Libraries

In [28]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
%config InlineBackend.figure_format = 'retina'
import seaborn as sns

### Importing cleaned dataset

In [2]:
# Importing cleaned dataset and dropping old index

retail = pd.read_csv('./dataset/retail.csv')
retail = retail.drop('Unnamed: 0', axis=1)
retail.head()

Branch YearMonth  Weekday  Hour            Product line  Unit price  \
0      A   2019-01        5    13       Health and beauty       74.69   
1      C   2019-03        4    10  Electronic accessories       15.28   
2      A   2019-03        6    13      Home and lifestyle       46.33   
3      A   2019-01        6    20       Health and beauty       58.22   
4      A   2019-02        4    10       Sports and travel       86.31   

   Quantity     Total  Rating  
0         7  548.9715     9.1  
1         5   80.2200     9.6  
2         7  340.5255     7.4  
3         8  489.0480     8.4  
4         7  634.3785     5.3

### First insights

In [8]:
retail.describe()

Weekday         Hour   Unit price     Quantity       Total  \
count  1000.000000  1000.000000  1000.000000  1000.000000  991.000000   
mean      3.032000    14.910000    55.672130     5.510000  316.604752   
std       1.973543     3.186857    26.494628     2.923431  237.709589   
min       0.000000    10.000000    10.080000     1.000000   10.678500   
25%       1.000000    12.000000    32.875000     3.000000  122.167500   
50%       3.000000    15.000000    55.230000     5.000000  252.756000   
75%       5.000000    18.000000    77.935000     8.000000  464.667000   
max       6.000000    20.000000    99.960000    10.000000  951.825000   

           Rating  
count  1000.00000  
mean      6.97270  
std       1.71858  
min       4.00000  
25%       5.50000  
50%       7.00000  
75%       8.50000  
max      10.00000

- The mean is most of the time closed to the median which shows that the data is concentrated around the mean.
- Unit price (in USD): between 10.08 to 99.96 with an average of 55.67
- Quantity: between 1 and 10 with a median of 5 - the buyers don't buy many articles at once
- Total spent (in USD): between 10ish to 951.82 with an average around 316
- Ratings: between 4 and 10 with a mean of 6.97

In [11]:
retail.columns

Index(['Branch', 'YearMonth', 'Weekday', 'Hour', 'Product line', 'Unit price',
       'Quantity', 'Total', 'Rating'],
      dtype='object')

### 1. Can we see different customer behavior depending on the branches?

In [32]:
# Checking distribution


In [17]:
branch_a = retail.loc[(retail.Branch == 'A')]

In [18]:
branch_a

Branch YearMonth  Weekday  Hour            Product line  Unit price  \
0        A   2019-01        5    13       Health and beauty       74.69   
2        A   2019-03        6    13      Home and lifestyle       46.33   
3        A   2019-01        6    20       Health and beauty       58.22   
4        A   2019-02        4    10       Sports and travel       86.31   
6        A   2019-02        0    14  Electronic accessories       68.84   
..     ...       ...      ...   ...                     ...         ...   
990      A   2019-03        4    19      Food and beverages       56.56   
992      A   2019-03        6    20  Electronic accessories       58.03   
997      A   2019-02        5    13      Food and beverages       31.84   
998      A   2019-02        4    15      Home and lifestyle       65.82   
999      A   2019-02        0    13     Fashion accessories       88.34   

     Quantity     Total  Rating  
0           7  548.9715     9.1  
2           7  340.5255     7.4  
3           8  489.0480     8.4  
4           7  634.3785     5.3  
6           6  433.6920     5.8  
..        ...       ...     ...  
990         5  296.9400     4.5  
992         2  121.8630     8.8  
997         1   33.4320     7.7  
998         1   69.1110     4.1  
999         7  649.2990     6.6  

[340 rows x 9 columns]